hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientReliableTopicProxy.java

```diff
@@ -197,6 +213,125 @@ public LocalTopicStats getLocalTopicStats() {
         throw new UnsupportedOperationException("Locality is ambiguous for client!");
     }
 
+    @Override
+    public void publishAll(@Nonnull Collection<? extends E> payload) {
+        checkNotNull(payload, NULL_MESSAGE_IS_NOT_ALLOWED);
+        checkNoNullInside(payload, NULL_MESSAGE_IS_NOT_ALLOWED);
+
+        try {
+            List<ReliableTopicMessage> messages = payload.stream()
+                    .map(m -> new ReliableTopicMessage(toData(m), null))
+                    .collect(Collectors.toList());
+            switch (overloadPolicy) {
+                case ERROR:
+                    long sequenceId = ringbuffer.addAllAsync(messages, OverflowPolicy.FAIL).toCompletableFuture().get();
+                    if (sequenceId == -1) {
+                        throw new TopicOverloadException("Failed to publish messages: " + payload + " on topic:" + getName());
+                    }
+                    break;
+                case DISCARD_OLDEST:
+                    ringbuffer.addAllAsync(messages, OverflowPolicy.OVERWRITE).toCompletableFuture().get();
+                    break;
+                case DISCARD_NEWEST:
+                    ringbuffer.addAllAsync(messages, OverflowPolicy.FAIL).toCompletableFuture().get();
+                    break;
+                case BLOCK:
+                    long timeoutMs = INITIAL_BACKOFF_MS;
+                    for (; ; ) {
+                        long result = ringbuffer.addAllAsync(messages, OverflowPolicy.FAIL).toCompletableFuture().get();
+                        if (result != -1) {
+                            break;
+                        }
+
+                        MILLISECONDS.sleep(timeoutMs);
+                        timeoutMs *= 2;
+                        if (timeoutMs > MAX_BACKOFF) {
+                            timeoutMs = MAX_BACKOFF;
+                        }
+                    }
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unknown overloadPolicy:" + overloadPolicy);
+            }
+        } catch (RuntimeException e) {
+            throw e;
+        } catch (Exception e) {
+            throw (RuntimeException) peel(e, null,
+                    "Failed to publish messages: " + payload + " to topic:" + getName());
+        }
+    }
+
+    @Override
+    public CompletionStage<Void> publishAllAsync(@Nonnull Collection<? extends E> payload) {
+        checkNotNull(payload, NULL_MESSAGE_IS_NOT_ALLOWED);
+        checkNoNullInside(payload, NULL_MESSAGE_IS_NOT_ALLOWED);
+        InternalCompletableFuture<Void> returnFuture = new InternalCompletableFuture<>();
+
+        List<ReliableTopicMessage> messages = payload.stream()
```

Comment:
Minor: this was previously in the try-catch block and was wrapped in case serialization failed (I think) so maybe we can follow the same approach here. Not sure if we should complete the future or throw it to the caller. Maybe fail-fast and throwing it directly is the better option here.

Reply:
I've changed the code in the same way like `publishAll`

Reply:
I don't see the change, have you pushed it?

Reply:
Now it's pushed.